### ISSUE TITLE
Cannot submit a new list to original adapter when using DataBinding

### ISSUE THREAD

**Issue by ericktijerou** [ericktijerou/koleton#14]
Reported by @TurKurT656

I'm using DataBinding library to set data to the RecyclerView:

```
@BindingAdapter("binding:setData")
fun <T> RecyclerView.setRecyclerViewData(data: List<T>?) {
    data?.let {
         if (adapter is MyAdapter<*>)
            (adapter as MyAdapter<T>).submitList(it)
    }
}
```
so I think when data is on loading state the library changes RecyclerView to KoletonView and because of that the BindingAdapter function never invokes. I've updated the sample project to demonstrate the issue.

**Comment by ericktijerou** [comment]
@TurKurT656 I found the problem:

When the recyclerview's skeleton is showing, the original adapter is replaced by a `KoletonAdapter`. After, when hideSkeleton is called the original adapter is restored. 

So, if you try to get the adapter from`recyclerview.adapter` before hiding the skeleton returns a KoletonAdapter. _In other cases the problem doesn't exist because you have the adapter instance._

I've created a temporary fix for this issue, try out `0.7.6` version. You should use `afterHideSkeleton ` lambda expression in your `BindingAdapter` .

```
@BindingAdapter("setData")
fun RecyclerView.setRecyclerViewData(data: List<MyItem>?) {
    afterHideSkeleton {
        data?.let { if (adapter is MyAdapter) (adapter as MyAdapter).submitList(it) }
    }
}
```

I will think of a better way and of course you are free to create a pull request which I would gladly review it.

Keep in touch!

**Comment by TurKurT656** [comment]
I've tested this. everything works fine.
Tnx a lot
